src/extension.ts

```diff
@@ -199,10 +198,54 @@ export async function activate(ctx: vscode.ExtensionContext) {
           break;
       }
     }
+  }
 
-    if (configureOnOpen === true) {
-      runFirstTask("reconfigure");
+  const downloadLanguageServerKey = "downloadLanguageServer";
+  const downloadLanguageServer = extensionConfiguration(downloadLanguageServerKey);
+  const languageServerKey = "languageServer"
+  const lsKey = extensionConfiguration(languageServerKey);
+  const possibleServers = {
+    "Swift-MesonLSP": SwiftMesonLspLanguageClient
+  }
+  const client = new possibleServers[lsKey](ctx);
+  const canDownload = client.canDownloadLanguageServer();
+  let registerTooling = true;
+  if (canDownload && downloadLanguageServer == "ask") {
+    enum Options {
+      yes = "Yes",
+      no = "Not this time",
+      never = "Never"
     }
+    const response = await vscode.window.showInformationMessage(
+      `Would you like to download a language server? (${client.repoURL})`,
+      ...Object.values(Options)
+    );
+
+    switch (response) {
+      case Options.no:
+        extensionConfigurationSet(downloadLanguageServerKey, "ask", vscode.ConfigurationTarget.Global);
+        break;
+
+      case Options.never:
+        extensionConfigurationSet(downloadLanguageServerKey, "never", vscode.ConfigurationTarget.Global);
+        break;
+
+      case Options.yes:
+        registerTooling = false;
+        client.setupLanguageServer();
+        extensionConfigurationSet(downloadLanguageServerKey, "yes", vscode.ConfigurationTarget.Global);
+        break;
+    }
+  } else if (canDownload && downloadLanguageServer == "yes") {
+    registerTooling = false;
+    client.setupLanguageServer();
+  }
+
+  if (registerTooling) {
+    activateLinters(root, ctx);
+    activateFormatters(ctx);
+  } else {
+    getOutputChannel().appendLine("Not enabling muon linter/formatter as language server is active!");
```

Comment:
```suggestion
    getOutputChannel().appendLine("Not enabling muon linter/formatter as language server is active!");
    getOutputChannel().appendLine("Not enabling the muon linter/formatter because the the language server is active!");
```